backend/src/main/java/com/htecgroup/SkyNest/exception/UserException.java

```diff
@@ -0,0 +1,11 @@
+package com.htecgroup.SkyNest.exception;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@Getter
+@AllArgsConstructor
+public class UserException extends RuntimeException {
```

Comment:
Maybe consider the option to generise this as well, since there will be different custom exceptions in this project, maybe add a parent abstract class and use that one in the ControllerAdvice class, it may save you lot of duplicate code. If you feel it is not the time for it, leave it for later with a TODO note

Reply:
I think I get it, make it so we can throw an exception with a custom message / HttpStatus that might not make sense often, so it would be confusing to have it in the UserExceptionTypes. Will look into it